parser_library/src/workspaces/workspace_configuration.cpp

```diff
@@ -853,18 +926,42 @@ const program* workspace_configuration::get_program(const utils::resource::resou
 std::pair<const program*, bool> workspace_configuration::get_program_normalized(
     const utils::resource::resource_location& file_location_normalized) const
 {
+    constexpr auto try_retrieve_program = [](const program_related_details& p_rel_details,
+                                              bool exact) -> std::pair<const program*, bool> {
+        if (const auto& tagged_pgm = std::get_if<tagged_program>(&p_rel_details); tagged_pgm)
+            return { &tagged_pgm->pgm, exact };
+        else
+            return { nullptr, false };
+    };
+
     // direct match
-    if (auto program = m_exact_pgm_conf.find(file_location_normalized); program != m_exact_pgm_conf.cend())
-        return { &program->second.pgm, true };
+    if (auto pgm_details = m_exact_pgm_conf.find(file_location_normalized); pgm_details != m_exact_pgm_conf.cend())
+        return try_retrieve_program(pgm_details->second, true);
 
-    for (const auto& [program, pattern] : m_regex_pgm_conf)
+    for (const auto& [pgm_details, pattern] : m_regex_pgm_conf)
     {
         if (std::regex_match(file_location_normalized.get_uri(), pattern))
-            return { &program.pgm, false };
+            return try_retrieve_program(pgm_details, false);
     }
+
     return { nullptr, false };
 }
 
+const workspace_configuration::missing_pgroup_details* workspace_configuration::get_missing_pgroup_details_normalized(
```

Comment:
As there does not seem to be a version that does not take the non-normalize file_location, I'd just drop the normalized suffix.